docs/summer-of-code-ideas/summer-of-code-reports/gsoc-2020-andreperezmaselco.md

```diff
@@ -0,0 +1,203 @@
+  <tr>
+    <th>
+      Reference shader
+    </th>
+    <th>
+      Variant shader
+    </th>
+  </tr>
+  <tr>
+    <td>
+      <pre>
+        <code>
+      OpCapability Shader
+ %1 = OpExtInstImport "GLSL.std.450"
+      OpMemoryModel Logical GLSL450
+      OpEntryPoint Vertex %10 "main"
+ %2 = OpTypeFloat 32
+ %3 = OpTypeVoid
+ %4 = OpTypeFunction %3
+ %5 = OpTypeVector %2 2
+ %6 = OpConstant %2 1
+ %7 = OpConstant %2 2
+ %8 = OpConstant %2 3
+ %9 = OpConstantComposite %5 %6 %7
+%10 = OpFunction %3 None %4
+%11 = OpLabel
+<strong>%12 = OpVectorTimesScalar %5 %9 %8</strong>
+      OpReturn
+      OpFunctionEnd
+        </code>
+      </pre>
+    </td>
+    <td>
+      <pre>
+        <code>
+      OpCapability Shader
+ %1 = OpExtInstImport "GLSL.std.450"
+      OpMemoryModel Logical GLSL450
+      OpEntryPoint Vertex %10 "main"
+ %2 = OpTypeFloat 32
+ %3 = OpTypeVoid
+ %4 = OpTypeFunction %3
+ %5 = OpTypeVector %2 2
+ %6 = OpConstant %2 1
+ %7 = OpConstant %2 2
+ %8 = OpConstant %2 3
+ %9 = OpConstantComposite %5 %6 %7
+%10 = OpFunction %3 None %4
+%11 = OpLabel
+<strong>%13 = OpCompositeExtract %2 %9 0
+%14 = OpFMul %2 %13 %8
+%15 = OpCompositeExtract %2 %9 1
+%16 = OpFMul %2 %15 %8
+%12 = OpCompositeConstruct %5 %14 %16</strong>
+      OpReturn
+      OpFunctionEnd
+        </code>
+      </pre>
+    </td>
+  </tr>
+</table>
+
+This project was concerned to use the metamorphic testing approach to extend *spirv-fuzz* by adding new transformations to cover features of *SPIR-V* that are not yet very widely-used, but that will be in the future.
+Also, the project used tools such as [CLSmith](https://github.com/ChrisLidbury/CLSmith) and [clspv](https://github.com/google/clspv) to improve *spirv-fuzz*.
+
+## Development
+
+Transformations and fuzzer passes work together to modify the shader source code.
+The fuzzer pass iterates over the *SPIR-V* module and decides whether a transformation will be applied.
+Below are the transformations, fuzzer passes and improvements developed during the program.
+
+### Transformations and Fuzzer Passes
+
+- [#3205](https://github.com/KhronosGroup/SPIRV-Tools/pull/3205): Swap Commutable Operands
+
+- [#3211](https://github.com/KhronosGroup/SPIRV-Tools/pull/3211): Toggle Access Chain Instruction
+
+- [#3336](https://github.com/KhronosGroup/SPIRV-Tools/pull/3336): Adjust Branch Weights
+
+- [#3359](https://github.com/KhronosGroup/SPIRV-Tools/pull/3359): Push Ids Through Variable
+
+- [#3402](https://github.com/KhronosGroup/SPIRV-Tools/pull/3402): Replace Linear Algebra Instruction
+
+- [#3412](https://github.com/KhronosGroup/SPIRV-Tools/pull/3412): Add Vector Shuffle Instruction
+
+- [#3439](https://github.com/KhronosGroup/SPIRV-Tools/pull/3439): Add Image Sample Unused Components
+
+- [#3597](https://github.com/KhronosGroup/SPIRV-Tools/pull/3597): Make Vector Operations Dynamic
+
+- [#3517](https://github.com/KhronosGroup/SPIRV-Tools/pull/3517): Inline Function
+
+- Add Bit Instruction Synonym (Work In Progress)
```

Comment:
You can of course do another PR after the deadline to add the link for this if you like.